internal/rhcs/cluster/clusterschema/sts.go

```diff
@@ -0,0 +1,163 @@
+package clusterschema
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	common2 "github.com/terraform-redhat/terraform-provider-rhcs/internal/rhcs/common"
+	"strings"
+)
+
+func StsFields() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"oidc_endpoint_url": {
+			Description: "OIDC Endpoint URL",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+		},
+		"oidc_config_id": {
+			Description: "OIDC Configuration ID",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"thumbprint": {
+			Description: "SHA1-hash value of the root CA of the issuer URL",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"role_arn": {
+			Description: "Installer Role",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"support_role_arn": {
+			Description: "Support Role",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"instance_iam_roles": {
+			Description: "Instance IAM Roles",
+			Type:        schema.TypeList,
+			MaxItems:    1,
+			Elem: &schema.Resource{
+				Schema: instanceRoleFields(),
+			},
+			Required: true,
+		},
+		"operator_role_prefix": {
+			Description: "Operator IAM Role prefix",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+	}
+}
+
+func instanceRoleFields() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"master_role_arn": {
+			Description: "Master/Controller Plane Role ARN",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"worker_role_arn": {
+			Description: "Worker Node Role ARN",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+	}
+}
+
+type Sts struct {
+	// required
+	RoleARN            string          `tfsdk:"role_arn"`
+	SupportRoleArn     string          `tfsdk:"support_role_arn"`
+	InstanceIAMRoles   InstanceIAMRole `tfsdk:"instance_iam_roles"`
+	OperatorRolePrefix string          `tfsdk:"operator_role_prefix"`
+
+	// optional
+	OIDCEndpointURL *string `tfsdk:"oidc_endpoint_url"`
+	OIDCConfigID    *string `tfsdk:"oidc_config_id"`
+
+	// computed
+	Thumbprint string `tfsdk:"thumbprint"`
+}
+
+type InstanceIAMRole struct {
+	MasterRoleARN string `tfsdk:"master_role_arn"`
+	WorkerRoleARN string `tfsdk:"worker_role_arn"`
+}
+
+func ExpandInstanceIamRolesFromInterface(i interface{}) InstanceIAMRole {
+	l := i.([]interface{})
+	if len(l) == 0 || l[0] == nil {
+		return InstanceIAMRole{}
+	}
+
+	instanceRoleMap := l[0].(map[string]interface{})
+	return InstanceIAMRole{
+		MasterRoleARN: instanceRoleMap["master_role_arn"].(string),
+		WorkerRoleARN: instanceRoleMap["worker_role_arn"].(string),
+	}
+}
+
+func ExpandStsFromResourceData(resourceData *schema.ResourceData) *Sts {
+	list, ok := resourceData.GetOk("sts")
+	if !ok {
+		return nil
+	}
+
+	return ExpandStsFromInterface(list)
+}
+
+func ExpandStsFromInterface(i interface{}) *Sts {
```

Comment:
maybe we can changes all expands to use https://github.com/mitchellh/mapstructure ?
I assume it always has map[string]interface? so it can be useful and pretty easy to use something more generic?